free-4.md

```diff
@@ -0,0 +1,92 @@
+good way to meet their needs.
+
+
+### What's the need?
+_What need is met by your idea? Who are you helping? What is that person's
+experience like now? What would their experience be like if you could help 
+them?_
+
+This project would serve weird people (me) who want the ability to transform text into different encodings. If a person were writing a message in Morse Code and realized that they wanted to see what it looked like in Braille, this language would be able to meet their needs.
+
+The idea meets the needs of people who enjoy encodings but don't want to take the time translating between them. It would be especially useful for events like [puzzlehunts](https://en.wikipedia.org/wiki/Puzzlehunt) in which participants solve puzzles. Some of these puzzles involve multiple ciphers which the average person can't translate on the fly. With this language, the user would only have to input the encoded text as well as the language of encoding/decoding and the user would understand what the encoded text was saying.
+
+
+### Why a language?
+_Why is a DSL appropriate for your user(s)? How does it address the need?_
+
+A DSL would be appropriate for users because it would give them a lot of options when encoding or decoding text. The user would not necessarily need to translate to/from English and could select from a wide variety of encodings. It could also include the functionality of listing how a string would appear in multiple different encodings, something which would address a user's need for more information and could help with time constraints in encoding puzzles.
+
+
+### Why you?
+_What excites you about this idea? How did you come up with it?_
+
+I have participated in a lot of puzzle hunts and I really enjoy learning new encodings. In the past, teammates who don't have encodings memorized take a while to decode them. If they use online solvers, they need to use a different solver for each type of encoding. I came up with this idea in the interst of consolidating many different decoding tools into a single resource.
+
+
+### Domain
+_Describe the project's domain in five words._
+
+Translating between all the encodings.
+
+
+### Interface (syntax)
+_How might the user interact with the language? What does programming look 
+like? Why is this the right way to interact with the problem domain?_ 
+
+The interface could be either a CLI or a GUI. I could see this as a web app resembling [Google Translate](https://translate.google.com/) where the user would specify two encodings. There could also be options in the GUI to list the text in all available encodings. I think that it might be easier to add functionality to the tool if it were a command line application. This would limit the user base to people who are comfortable with using the command line. I think that this could provide a more reasonable and extensible method of interaction for users. I could see a sample command being something like: 'encoding1 encoding2 textToBeTranslated'
```

Comment:
I agree a command line interface would probably be a lot nicer in terms of richness in functionality. I think by making it a command line interface, it would also entail more language design choices rather than sticking everything into a GUI.